dvc/repo/__init__.py

```diff
@@ -195,11 +195,14 @@ def _collect_dir_cache(
 
         return ret
 
+    def _out_missing_version_info(self, out):
+        return self.cache.local.PARAM_CHECKSUM not in out.info.keys()
+
     def _collect_used_cache(
         self, out, branch=None, remote=None, force=False, jobs=None
     ):
-        if not out.use_cache or not out.info:
-            if not out.info:
+        if not out.use_cache or self._out_missing_version_info(out):
```

Comment:
You should pop PARAM_REMOVE here https://github.com/iterative/dvc/blob/master/dvc/output/__init__.py#L75 and then you won't need to modify this.